src/MessagePack/Resolvers/DynamicObjectResolver.cs

```diff
@@ -314,10 +312,10 @@ namespace MessagePack.Internal
 {
     internal static class DynamicObjectTypeBuilder
     {
-#if !UNITY_2018_3_OR_NEWER
-        private static readonly Regex SubtractFullNameRegex = new Regex(@", Version=\d+.\d+.\d+.\d+, Culture=\w+, PublicKeyToken=\w+", RegexOptions.Compiled);
+#if NETFRAMEWORK || NETSTANDARD2_0
+        private static readonly Regex SubtractFullNameRegex = new(@", Version=\d+.\d+.\d+.\d+, Culture=\w+, PublicKeyToken=\w+", RegexOptions.Compiled);
 #else
-        static readonly Regex SubtractFullNameRegex = new Regex(@", Version=\d+.\d+.\d+.\d+, Culture=\w+, PublicKeyToken=\w+");
+        private static readonly Regex SubtractFullNameRegex = new(@", Version=\d+.\d+.\d+.\d+, Culture=\w+, PublicKeyToken=\w+", System.Runtime.CompilerServices.RuntimeFeature.IsDynamicCodeSupported ? RegexOptions.Compiled : RegexOptions.None);
```

Comment:
We already have a `static readonly` field somewhere that is initialized with `IsDynamicCodeSupported`, taking the `#if` into account. Can we make that `internal` and reuse that throughout the library? That should cut down on our need for `#if` in this and other places.

Reply:
I understand. I'll reuse it as you said.